src/formatDate.js

```diff
@@ -8,7 +8,48 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldYearIndex = getIndex(fromFormat);
+  const newYearIndex = getIndex(toFormat);
+  const oldYearLength = fromFormat[oldYearIndex].length;
+  const newYearLength = toFormat[newYearIndex].length;
+  const oldJoin = fromFormat[3];
+  const newJoin = toFormat[3];
+  const newDate = date.split(oldJoin);
```

Comment:
The variable name `newDate` is misleading since it's not a new date, but rather an array of the date parts. A more suitable name could be `dateParts`.